framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,726 @@
+package org.checkerframework.common.signatureinference;
```

Comment:
Thanks for bringing up this issue.  I think the right approach is to forbid the use of the two flags simultaneously (Javier's second option).

Also, I don't think that a user will expect that code outside `@AnnotatedFor` won't be touched.  Rather, the point of using signature inference is to start or complete one's annotation work and to add annotations even where ones don't already exist.  Therefore, I don't think the second subtlety needs to be highlighted.